src/firefly/js/templates/lightcurve/basic/BasicMissionOptions.js

```diff
@@ -1,55 +1,91 @@
 import React, {Component, PropTypes} from 'react';
 import sCompare from 'react-addons-shallow-compare';
-import {get, has, isEmpty, set} from 'lodash';
+import {flux} from '../../../Firefly.js';
+import {get, has, isEmpty, set, defer} from 'lodash';
 import {FieldGroup} from '../../../ui/FieldGroup.jsx';
 import {ValidationField} from '../../../ui/ValidationField.jsx';
 import {SuggestBoxInputField} from '../../../ui/SuggestBoxInputField.jsx';
 import {makeFileRequest} from '../../../tables/TableUtil.js';
-import {getColumnIdx, getTblById, smartMerge} from '../../../tables/TableUtil.js';
-import {sortInfoString} from '../../../tables/SortInfo.js';
+import {getColumnIdx, smartMerge} from '../../../tables/TableUtil.js';
 import {ReadOnlyText, getTypeData} from '../LcUtil.jsx';
-import {LC, getViewerGroupKey} from '../LcManager.js';
-import {getMissionName, coordSysOptions} from '../LcConverterFactory.js';
+import {LC, getViewerGroupKey, makeRawTableRequestByColumnChange} from '../LcManager.js';
+import {getMissionName} from '../LcConverterFactory.js';
+import {getLayouInfo} from '../../../core/LayoutCntlr.js';
+import {sortInfoString} from '../../../tables/SortInfo.js';
 
 const labelWidth = 90;
-
+const NumTypes = ['double', 'd', 'long', 'l', 'int', 'i', 'float', 'f'];
+const CharTypes = ['char', 'c', 's', 'str'];
 
 export class BasicSettingBox extends Component {
     constructor(props) {
         super(props);
+        this.classifyColumns = (columns) => {
+            return isEmpty(columns) ? {} :
+                columns.reduce((prev, col) => {
+                    if (get(col, 'visibility', '') !== 'hidden') {
+                        if (NumTypes.includes(col.type)) {
+                            prev.numColumns.push(col.name);
+                        } else if (CharTypes.includes(col.type)) {
+                            prev.charColumns.push(col.name);
+                        }
+                    }
+                    return prev;
+                }, {charColumns: [], numColumns:[]});
```

Comment:
This logic should go into TableUtil.js.  It can be separated into getNumericColNames and getStringColNames.
It'll hide some of table's inner structures and it'll make the code much easier to read.  i.e. columns.filter(...).map(...)